pkg/operator/controller/route-metrics/controller.go

```diff
@@ -174,16 +174,14 @@ func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 	// NOTE: Even though the route admitted status should reflect validity of the namespace and route labelselectors, we still will validate
 	// the namespace and route labels as there are still edge scenarios where the route status may be inaccurate.
 
-	// Get the Namespace LabelSelector from the Ingress Controller.
-	var namespaceLabelSelector map[string]string
-	if ingressController.Spec.NamespaceSelector != nil {
-		namespaceLabelSelector = ingressController.Spec.NamespaceSelector.MatchLabels
+	// List all the Namespaces filtered by our ingress's Namespace selector.
+	namespaceSelector, err := metav1.LabelSelectorAsSelector(ingressController.Spec.NamespaceSelector)
+	if err != nil {
+		log.Error(err, "ingresscontroller has an invalid namespace selector", "ingresscontroller", ingressController.Name)
```

Comment:
It might be helpful to log `ingressController.Spec.NamespaceSelector)` as well... but then again, the operator's ingress controller will be logging it since `desiredRouterDeployment` returns an error if it cannot parse it.

Reply:
Added it just for completeness